crates/ty_python_semantic/src/types/constraints.rs

```diff
@@ -81,7 +81,7 @@ use crate::Db;
 use crate::types::{BoundTypeVarInstance, IntersectionType, Type, UnionType};
 
 /// Encodes the constraints under which a type property (e.g. assignability) holds.
-pub(crate) trait Constraints<'db>: Clone + Sized {
+pub(crate) trait Constraints<'db>: Clone + Sized + std::fmt::Debug {
```

Comment:
I wanted to do a `dbg!()` call and it didn't work because a `Constraints` instance wasn't guaranteed to implement `Debug`. This improves debugabbility.